drivers/flash/spi_nor.c

```diff
@@ -468,30 +530,34 @@ static int enter_dpd(const struct device *const dev)
 static int exit_dpd(const struct device *const dev)
 {
 	int ret = 0;
+	const struct spi_nor_config *cfg = dev->config;
 
-	if (IS_ENABLED(DT_INST_PROP(0, has_dpd))) {
+	if (cfg->dpd_exist) {
 		delay_until_exit_dpd_ok(dev);
 
-#if DT_INST_NODE_HAS_PROP(0, dpd_wakeup_sequence)
-		/* Assert CSn and wait for tCRDP.
-		 *
-		 * Unfortunately the SPI API doesn't allow us to
-		 * control CSn so fake it by writing a known-supported
-		 * single-byte command, hoping that'll hold the assert
-		 * long enough.  This is highly likely, since the
-		 * duration is usually less than two SPI clock cycles.
-		 */
-		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RDID);
-
-		/* Deassert CSn and wait for tRDP */
-		k_sleep(K_MSEC(T_RDP_MS));
-#else /* DPD_WAKEUP_SEQUENCE */
-		ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RDPD);
+#if ANY_INST_HAS_DPD_WAKEUP_SEQUENCE
+		if (cfg->dpd_wakeup_sequence_exist) {
+			/* Assert CSn and wait for tCRDP.
+			 *
+			 * Unfortunately the SPI API doesn't allow us to
+			 * control CSn so fake it by writing a known-supported
+			 * single-byte command, hoping that'll hold the assert
+			 * long enough.  This is highly likely, since the
+			 * duration is usually less than two SPI clock cycles.
+			 */
+			ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RDID);
 
-		if (ret == 0) {
-#if DT_INST_NODE_HAS_PROP(0, t_exit_dpd)
-			k_sleep(K_MSEC(T_RES1_MS));
+			/* Deassert CSn and wait for tRDP */
+			k_sleep(K_MSEC(cfg->t_rdp_ms));
+		} else {
+			ret = spi_nor_cmd_write(dev, SPI_NOR_CMD_RDPD);
+			if (ret == 0) {
+#if ANY_INST_HAS_T_EXIT_DPD
+				if (cfg->dpd_exist) {
+					k_sleep(K_MSEC(cfg->t_exit_dpd));
+				}
```

Comment:
You can just probably take the entire `if (ret == 0)` block as conditional; I think that some compilers/static analysis may not be happy with empty block in `if`.